ui.frontend/src/constants.js

```diff
@@ -42,6 +42,48 @@ export const Constants = {
      */
     FORM_CONTAINER_DATA_ATTRIBUTE: "adaptiveformcontainerPath",
 
+    /**
+     * data attribute to mark the dragged component valid or invalid.
+     * value true for valid
+     * value false for invalid
+     */
+    VALID : "data-cmp-valid",
+
+    /**
+     * data attribute to mark the dragged component enabled or disabled.
+     * value true for enabled
+     * value false for disabled
+     */
+    ENABLED : "data-cmp-enabled",
+
+    /**
+     * data attribute to mark the dragged component visible or invisible.
+     * value true for visible
+     * value false for invisible
+     */
+    VISIBLE : "data-cmp-visible",
+
+    /**
+     * data attribute to mark the dragged component active or inactive.
+     * value true for active
+     * value false for inactive
+     */
+    ACTIVE : "data-cmp-active",
+
+    /**
+     * aria attribute to mark the dragged component disabled.
+     */
+    ARIA_DISABLED : "aria-disabled",
+
+    /**
+     * aria attribute to mark the dragged component hidden.
+     */
+    ARIA_HIDDEN : "aria-hidden",
+
+    /**
+     * aria attribute to mark the dragged component invalid.
+     */
+    ARIA_INVALID : "aria-invalid",
```

Comment:
I would suggest not to define constants for already public strings. If I can't remember `aria-disabled` I would not remember `Constants.ARIA_DISABLED`

Reply:
but we can atleast reuse it in our code

Reply:
`aria-disabled` vs `Constants.ARIA_DISABLED`. It would have made sense if there was some type checking.